src/ui_interface.h

```diff
@@ -72,43 +75,69 @@ class CClientUIInterface
         MSG_ERROR = (ICON_ERROR | BTN_OK | MODAL)
     };
 
+#define ADD_SIGNALS_DECL_WRAPPER(signal_name)                                              \
+    boost::signals2::connection signal_name##_connect(std::function<signal_name##Sig> fn); \
+    void signal_name##_disconnect(std::function<signal_name##Sig> fn);
+
     /** Show message box. */
-    boost::signals2::signal<bool (const std::string& message, const std::string& caption, unsigned int style), boost::signals2::last_value<bool> > ThreadSafeMessageBox;
+    bool  ThreadSafeMessageBox          (const std::string& message, const std::string& caption, unsigned int style);
+    using ThreadSafeMessageBoxSig = bool(const std::string& message, const std::string& caption, unsigned int style);
+    ADD_SIGNALS_DECL_WRAPPER(ThreadSafeMessageBox);
```

Comment:
Maybe overkill but
```C++
#define ADD_SIGNALS_DECL_WRAPPER(signal_name, rtype, args...) \
    rtype signal_name(args);\
    using signal_name##Sig = rtype(args);\
    [...]
```
then e.g.
```C++
    ADD_SIGNALS_DECL_WRAPPER(ThreadSafeMessageBox, bool, const std::string& message, const std::string& caption, unsigned int style);
```

Reply:
Oh thanks! That makes the diff in the header file look a lot cleaner.